src/Repo_old.java

```diff
@@ -0,0 +1,804 @@
+        final String ca = props.getProperty(PROP_CA, null);
+        final ServerInfo info = new ServerInfo();
+        String userName = props.getProperty(PROP_USER, null);
+        String password = props.getProperty(PROP_PASSWORD, null);
+        final Server server;
+
+        if (userName == null) {
+            userName = url.getUserName();
+        }
+        if (password == null) {
+            password = url.getPassword();
+        }
+        if (ca != null) {
+            if (Boolean.parseBoolean(ca)) {
+                info.setAutoLocateCacheAgent(true);
+            } else {
+                final int ind = ca.indexOf(':');
+
+                if (ind == -1) {
+                    info.setMPXCacheAgentAddress(ca);
+                } else {
+                    info.setMPXCacheAgentAddress(ca.substring(0, ind));
+                    info.setMPXCacheAgentPort(Integer.parseInt(ca
+                            .substring(ind + 1)));
+                }
+            }
+        }
+
+        info.setHost(host);
+        info.setPort(port);
+        info.setCompression(true);
+        info.setConnectionType(protocol);
+        info.setMPXCacheAgentThreadCount(Integer.parseInt(props.getProperty(
+                PROP_CATHREADS, PROP_DEFAULT_CATHREADS)));
+        info.setEnableCacheAgentForFileContent(ca != null);
+        info.setEnableCacheAgentForObjectProperties(ca != null);
+        server = new Server(info);
+        server.connect();
+
+        if ((userName != null) && (password != null)) {
+            if (server.logOn(userName, password) == 0) {
+                throw new ConfigurationException("Failed to login to " + host
+                        + ':' + port + " as user " + userName);
+            }
+            cacheLogOnCredentials(server, userName, password);
+        } else if (autoLogOn(server) == 0) {
+            final CredentialHelper h = getGit().getCredentialHelper(
+                    getProtocol(url), host, userName);
+
+            if (h != null) {
+                final boolean[] approved = new boolean[1];
+                h.getCredentials(new CredentialCallBack() {
+                    @Override
+                    public boolean approve(final String user,
+                            final String password) {
+                        try {
+                            if (server.logOn(user, password) != 0) {
+                                cacheLogOnCredentials(server, user, password);
+                                props.setSessionProperty(PROP_USER, user);
+                                props.setSessionProperty(PROP_PASSWORD,
+                                        password);
+                                return approved[0] = true;
+                            } else {
+                                return approved[0] = false;
+                            }
+                        } catch (final LogonException ex) {
+                            return approved[0] = false;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)